### PR TITLE
Add support for synchronous USART mode

### PR DESCRIPTION
**Pull request by hannobraun**

### CODE REVIEW

File: src/usart/peripheral.rs [hunk 1]
```diff
@@ -133,18 +127,31 @@ where
...
+    fn configure<C>(&mut self, syscon: &mut syscon::Handle)
+    where
+        C: ClockSource,
+    {
+        syscon.enable_clock(&self.usart);
+        C::select(&self.usart, syscon);
+
+        // According to the user manual, section 13.6.1, we need to make sure
+        // that the USART is not sending or receiving data before writing to
+        // CFG, and that it is disabled. We statically know that it is disabled
```
Comment (david-sawatzke) [inline src/usart/peripheral.rs:247]: Doesn't this comment belong to the `cfg.modify` blocks above?

File: src/usart/peripheral.rs [hunk 2]
```diff
@@ -89,55 +87,92 @@ where
...
+    ///
+    /// # Limitations
+    ///
+    /// For USART to function correctly, the UARTFRG reset must be cleared. This
+    /// is the default, so unless you have messed with those settings, you
+    /// should be good.
+    ///
+    /// # Examples
+    ///
+    /// Please refer to the [module documentation] for a full example.
```
Comment (david-sawatzke) [inline src/usart/peripheral.rs:145]: Not sure this comment is accurate

File: src/usart/peripheral.rs [hunk 3]
```diff
@@ -146,9 +181,90 @@ where
...
```
Comment (david-sawatzke) [inline src/usart/peripheral.rs:202]: Same here

File: src/usart/state.rs [hunk 4]
```diff
@@ -32,6 +32,16 @@ impl Word for u16 {
     }
 }
 
+/// Indicates that a USART instance is operation in asynchronous mode
```
Comment (david-sawatzke) [inline src/usart/state.rs:35]:
```suggestion
/// Indicates that a USART instance is operating in asynchronous mode
```

File: src/usart/state.rs [hunk 5]
```diff
@@ -32,6 +32,16 @@ impl Word for u16 {
...
+///
+/// Used as a type parameter by `USART`.
+pub struct AsyncMode;
+
+/// Indicates that a USART instance is operation in synchronous mode
```
Comment (david-sawatzke) [inline src/usart/state.rs:40]:
```suggestion
/// Indicates that a USART instance is operating in synchronous mode
```

File: src/usart/peripheral.rs (hunk 1 above)
Comment (hannobraun) [reply src/usart/peripheral.rs:247]: That's totally right, good catch.

File: src/usart/peripheral.rs (hunk 2 above)
Comment (hannobraun) [reply src/usart/peripheral.rs:145]: Thanks, will remove.

File: src/usart/peripheral.rs (hunk 3 above)
Comment (hannobraun) [reply src/usart/peripheral.rs:202]: Thanks, will remove.

File: src/usart/state.rs (hunk 4 above)
Comment (hannobraun) [reply src/usart/state.rs]: Thanks. I could literally not see what the difference between my version and what you were suggesting here in the notification email. Good thing the web interface does the colored highlighting to help my tired brain out a bit.

**Comment by hannobraun** [comment]
Thanks for the review! Pushed new commits to address the comments. Should be good to go now.

**Pull request merged**